### PR TITLE
update recent object storage controllers to use consistent logging pattern

### LINKED ISSUE #7: Controllers should use consistent logging
We should decide on a single logging package and be consistent with it.  The logging package should support:
* concurrent usage
* configurable levels
* log file rollover

`log.Printf` is not sufficient for these goals.

### PR DESCRIPTION
**Pull request by jbw976**
**Description of your changes:** This PR updates the most recent controllers so that all controllers in the repo are using the consistent logging pattern established in #354.

There is also a commit to update missed RBAC permissions for the new Azure object storage types. 
 The pull request template checklist has been updated with a reminder to update RBAC permissions with each PR.

**Which issue is resolved by this Pull Request:**
Resolves #7 

**Checklist:**
- [x] Documentation has been updated, if necessary.
- [x] Pending release notes updated with breaking and/or notable changes, if necessary.
- [x] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [x] Code generation (`make generate`) has been run to update object specifications, if necessary.
- [x] CRD manifests generation (`make manifests`) has been run to update CRD manifests yaml file specifications, if necessary.
- [x] Update dependencies (`make vendor`) has been run to update `Gopkg.lock` file, if necessary
- [x] All related commits have been squashed to improve readability.
- [x] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](../blob/master/CONTRIBUTING.md#comments)

### CODE REVIEW

**Comment by negz** [comment]
> The pull request template checklist has been updated with a reminder to update RBAC permissions with each PR.

That will be useful! I usually forget to do this until I'm trying to use the new resource in Crossplane and wondering why it's not working.

File: pkg/controller/azure/storage/account/account.go [hunk 1]
```diff
@@ -74,6 +74,8 @@ var (
 	resultRequeue    = reconcile.Result{Requeue: true}
 	requeueOnSuccess = reconcile.Result{RequeueAfter: requeueAfterOnSuccess}
 	requeueOnWait    = reconcile.Result{RequeueAfter: requeueAfterOnWait}
+
+	log = logging.Logger.WithName("controller." + controllerName)
 )
 
 // Reconciler reconciles a GCP storage account acct
```
Comment (ichekrygin) [inline pkg/controller/azure/storage/account/account.go]: Not part of this PR, :) - GCP needs to be removed (I missed it in the original PR)

File: pkg/controller/azure/storage/account/account.go [hunk 2]
```diff
@@ -111,7 +113,7 @@ func Add(mgr manager.Manager) error {
 // Reconcile reads that state of the cluster for a Provider acct and makes changes based on the state read
 // and what is in the Provider.Spec
 func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	log.Printf("reconciling %s: %v", v1alpha1.AccountKindAPIVersion, request)
+	log.V(logging.Debug).Info("reconciling", "kind", v1alpha1.AccountKindAPIVersion, "request", request)
```
Comment (ichekrygin) [inline pkg/controller/azure/storage/account/account.go]: This is approximation of what a debug message will look like:
```
2019-03-20T00:16:41.846Z        INFO    controller.account.storage.azure.crossplane.io  reconciling        {"kind": "account.storage.azure.crossplane.io/v1alpha1", "request": "crossplane-system/my-test-account"}
```
This is to emphasize the point that perhaps we can improve on reducing redundancy in log messages.

File: pkg/apis/azure/storage/v1alpha1/types_test.go [hunk 3]
```diff
@@ -47,14 +47,14 @@ var (
 func TestMain(m *testing.M) {
 	err := SchemeBuilder.AddToScheme(scheme.Scheme)
 	if err != nil {
-		log.Fatal(err)
+		logcore.Fatal(err)
```
Comment (negz) [inline pkg/apis/azure/storage/v1alpha1/types_test.go]: I have a recollection that I'd replaced this pattern with a combination of a `fmt.Println` and an `os.Exit`, but I can't find any evidence to support that. 🤔

File: pkg/apis/azure/storage/v1alpha1/types_test.go [hunk 4]
```diff
@@ -17,7 +17,7 @@ limitations under the License.
 package v1alpha1
 
 import (
-	"log"
+	logcore "log"
```
[the remark this answers is not in the document]
Comment (jbw976) [reply pkg/apis/azure/storage/v1alpha1/types_test.go]: I wanted to continue using the existing `log.Fatal` functionality in this test, I didn't see the same thing offered from our new logging framework.  The import `log` conflicts with the new `log` variable newly declared elsewhere in this package in `pkg/apis/azure/storage/v1alpha1/account.go`.  So to keep the functionality and avoid the new naming conflict, I aliased the standard `log` import.  Open to other suggestions :)

File: pkg/apis/azure/storage/v1alpha1/types_test.go (hunk 3 above)
Comment (jbw976) [reply pkg/apis/azure/storage/v1alpha1/types_test.go]: I too was wondering what we did for `log.Fatal` in our tests, to avoid the naming conflict with the `log` variable we are now creating.  We are still using `log.Fatal` in other tests, but those don't have a `log` variable also declared, so there's no conflict.  We'll need a nice long term solution that bails out of the tests like `log.Fatal` does, but doesn't create create a conflict between the imported `log` package and the `log` variable.  Open to suggestions, but considering just leaving this `logcore` alias for this PR.

File: pkg/controller/azure/storage/account/account.go (hunk 2 above)
Comment (jbw976) [reply pkg/controller/azure/storage/account/account.go]: I'll open an issue to consider a more streamlined logging name scheme.  It's a good idea, but I don't have a great solution currently.

Comment (jbw976) [reply pkg/controller/azure/storage/account/account.go]: I've opened #388

File: pkg/apis/azure/storage/v1alpha1/types_test.go (hunk 3 above)
Comment (jbw976) [reply pkg/apis/azure/storage/v1alpha1/types_test.go]: I've opened #389 to track this

Comment (ichekrygin) [reply pkg/apis/azure/storage/v1alpha1/types_test.go]: I think I weigh in on: 
```go
err := SchemeBuilder.AddToScheme(scheme.Scheme)
	if err != nil {
		log.Error(err, "failed to add to scheme")
		os.Exit(1)
	}
```
